src/event-emitter.js

```diff
@@ -13,12 +13,14 @@ module.exports = function (classes){
       return msg;
     },
     /**
-     * Check if current request has an integer id
+     * Check if current request has an id adn it is of type integer (non fractional) or string.
+     *
      * @param {Object} request
      * @return {Boolean}
      */
     hasId : function (request){
-      return request && typeof request['id'] !== 'undefined' && /^\-?\d+$/.test(request['id']);
+      return request && typeof request['id'] !== 'undefined' &&
```

Comment:
it's missing checking against null (the `typeof !== 'undefined'` includes null, and all other types), so I think in the end it would be

```js
return request && typeof request['id'] !== 'undefined' && (typeof(request['id']) === 'number' && /^\-?\d+$/.test(request['id']) || typeof(request['id']) === 'string' || request['id'] === null);
```

Please notice that your typeof in string check is wrong. It's checking for the boolean if request['id'] === 'string' which will be always false, and it will always return truthy